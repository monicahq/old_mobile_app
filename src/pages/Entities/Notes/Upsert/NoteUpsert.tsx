import {Formik} from 'formik';
import React, {PureComponent} from 'react';
import {ScrollView, View} from 'react-native';

import {Navbar} from '@components';
import {I18n} from '@i18n';
import {INote} from '@models';
import {IPopAction} from '@navigator/NavigationService';
import {commonStyles} from '@theme';
import {NoteFormContainer} from './form/NoteFormContainer';
import {styles} from './NotesUpsert.style';

interface INotesUpsertProps {
  pop: IPopAction;
  updateNote: (note: INote) => void;
  contactId: number;
  note?: INote;
}

export class NoteUpsert extends PureComponent<INotesUpsertProps, {}> {
  private formRef = React.createRef<Formik>();

  public render() {
    const {pop, note} = this.props;
    const isEditing = !!note;

    return (
      <View style={commonStyles.flex}>
        <Navbar
          title={isEditing ? I18n.t('notes:edit') : I18n.t('notes:create')}
          onBack={pop}
          rightTitle={I18n.t('notes:save')}
          rightAction={this.handleDoneAction}
          rightIcon="done"
        />
        <ScrollView keyboardDismissMode="on-drag">
          <View style={styles.container}>
            <NoteFormContainer
              ref={this.formRef}
              note={note}
              onSuccess={this.onSuccess}
            />
          </View>
        </ScrollView>
      </View>
    );
  }

  private handleDoneAction = () => {
    this.formRef.current.submitForm();
  };

  private onSuccess = (noteValue: INote) => {
    const {note, updateNote, pop} = this.props;
    updateNote({
      ...note,
      ...noteValue,
    });
    pop();
  };
}
