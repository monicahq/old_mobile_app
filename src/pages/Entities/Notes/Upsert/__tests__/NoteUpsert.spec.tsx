import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {NoteUpsert} from '../NoteUpsert';

describe('Pages', () => {
  describe('Notes', () => {
    let pop;
    let updateNote;
    let contactId;
    let note;

    beforeEach(() => {
      pop = jest.fn();
      updateNote = jest.fn();
      contactId = 5;
      note = {body: 'c'};
    });

    it('should renders correctly', () => {
      const tree = shallow(
        <NoteUpsert
          pop={pop}
          updateNote={updateNote}
          contactId={contactId}
          note={note}
        />
      );
      expect(toJson(tree)).toMatchSnapshot();
    });
  });
});
