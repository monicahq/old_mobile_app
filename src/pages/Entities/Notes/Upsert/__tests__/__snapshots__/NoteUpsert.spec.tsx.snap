// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Notes should renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Navbar
    onBack={[MockFunction]}
    rightAction={[Function]}
    rightIcon="done"
    rightTitle="Done"
    title="Edit a note"
  />
  <ScrollView
    keyboardDismissMode="on-drag"
  >
    <Component
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#E2E4E5",
        }
      }
    >
      <[object Object]
        note={
          Object {
            "body": "c",
          }
        }
        onSuccess={[Function]}
      />
    </Component>
  </ScrollView>
</Component>
`;
